kolibri/core/content/utils/annotation.py

```diff
@@ -134,6 +135,93 @@ def fix_multiple_trees_with_id_one():
             )
 
 
+def update_num_coach_contents():
+    """
+    Function to set num_coach_content on topic trees that were imported before
+    the annotation would have been run.
+    """
+    bridge = Bridge(app_name=CONTENT_APP_NAME)
+
+    ContentNodeClass = bridge.get_class(ContentNode)
+
+    ContentNodeTable = bridge.get_table(ContentNode)
+
+    connection = bridge.get_connection()
+
+    child = ContentNodeTable.alias()
+
+    logger.info("Updating num_coach_content on existing channels")
+
+    # start a transaction
+
+    trans = connection.begin()
+
+    for channel_id in ChannelMetadata.objects.all().values_list("id", flat=True):
```

Comment:
I'm a little confused - does this operate on all channels, or just the ones that "that were imported before the annotation would have been run"?

Reply:
This operates on all channels, because we have no way of knowing (currently) when a channel was originally imported.

In 0.13 we will be able to use the version upgrade logic to restrict this to only happen on upgrades from before 0.12.4.